libraries/node/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+file(GLOB_RECURSE CONSENSUS_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/consensus/*.hpp)
+file(GLOB_RECURSE CONSENSUS_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/consensus/*.cpp)
+
+file(GLOB_RECURSE NETWORK_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/network/*.hpp)
+file(GLOB_RECURSE NETWORK_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/network/*.cpp)
```

Comment:
So network and consensus are the last parts that should be "logically" separated into the libs... I guess it was not possible due to some major problems, e.g. circular deps, etc... that is too difficult to fix ?

Reply:
First part is already discussed below I think.
I decided to make a structure like these to not mix all that files in common include/src directories and highlight that it should be a separate libs. But your proposal on structure is better except that `app_base` name :D

Reply:
yop feel free to come up with better naming just make sure it is something evoking that it consist of multiple libs....